src/records/dnsGeneric.ts

```diff
@@ -0,0 +1,18 @@
+import { Static, Boolean, String, Literal, Record, Union, Partial } from "runtypes";
+
+export const RecordTypesT = Literal("openatts");
+export const AlgorithmT = Union(Literal("dns-did"));
+export const VersionT = String;
+export const PublicKeyT = String;
```

Comment:
can use runtype constraints to validate the string format here

Reply:
done